Developing-an-Interactive-Educational-Platform-with-Directus.md

```diff
@@ -0,0 +1,374 @@
+
+We will create items in our courses;
+
+![](https://lh7-us.googleusercontent.com/xIC1ODpTfX5iSlRi5_QD0BUjYdt2Kf0Hskrz_GTfOKy_Ixi8wEGlJirkJx5U1HKqgv_nsNRGGwjys-IAOpawuiJFlNDZvj-ZpQ5pw192tpHoekr6F-s5JtKPG3B5R71lfO5boAVLhCsP2xhmnQeRsgg)
+
+If we navigate to: <https://interactive-educational-platform.directus.app/items/Courses>, we will see our raw data for our courses;
+
+![](https://lh7-us.googleusercontent.com/0py2EvtrdkF6LaREhqb20F_kuZ0jaukRHzgDj9nHByPfvnX7bpk28q-_m3uFLaBsM1nKS6h5LFS6emXsTdmyaWDJviRthHd45xYekfq2VvQEBBfr9c1AcChVuQEYt_hv_eh3GOtv46k2HSsVgdDBYKM)
+
+If we navigate to: <https://interactive-educational-platform.directus.app/items/Lesson>, we will see our raw data for our Lesson; we do have one lesson here, we could create more than one lessons if need be;
+
+![](https://lh7-us.googleusercontent.com/c2UkDhLS2PVgBiUm0zSGLgzqRcqXgqoC-RThI4s9QBYp00wW5Jvw6fPZ0w3IyIPycKbUVVRGSea1IqAuKWfPr0vkljU20xHEcpYCCtjZnOXo0GLZuSTE-FtObR7RbeCIoeLw58ZyL1gdPKSASqpRrIo)
+
+Replace the last words of the links above with the collection name so we can see our raw collections.  Well, the truth is you will not be able to do so, and here is why. You do not have permission to do so, that takes us to our next section, roles, and permissions. Let me break it down a bit, as of now whatever you are seeing is for the public eyes, and directus would not allow you see these contents until the right permissions are given.
+
+Roles and Permissions
+---------------------
+
+Roles in Directus can be easily created, navigate to the settings, and select Access Control,
+
+![](https://lh7-us.googleusercontent.com/iQrk7gZ_26xp2Q7H2AMPsUVHItEZTEg0cBX3yzh4UY1P2XJwTeTvtH0Hww-l4UOI75FJVpoiOzM8v8X7sGkAqgz7DmTw4BpQdgmVmpCw4ezd0Mq5VoTQYhhaSjz9aRx9lOiIXKfHMM9qFiYNRGndFxc)
+
+At the top right you will find a plus icon, click it create a role, and save. In our case we will create two roles the Students and the Instructors.
+
+![](https://lh7-us.googleusercontent.com/o_JTTKDATPEJ04qM9KEZ_J2EMu2gkOpYwsKBx6CKprmIRTflQNRgChJ5nNjDuPZNZSRfseIy-H3eA500cMPvuPAi6A8YT4I3W2sRlNovXRe4FBY12uv8PTtjQDDybu3lv3VQXb1jrSqiOU-rXhKCF3Y)
+
+By default we do have the Public and the Admin, we do not need to set permissions for the admin as it has all permissions.
+
+![](https://lh7-us.googleusercontent.com/BIdQat51I19KIs3vA3ahnTCxVHJcgqWvTBO9jILTB-k5dgxdghbBrRgboqA5SJB7h7gRWriBrkTCChPzLdlldrsX928dwb2WZzjf1u56KZKAoI9iSRB73VdCAulNXQ1S2subXi7pVMYaDp0-cewq9Bo)
+
+You want to be careful with those you give administrator roles to, as it has no restricted access, well for the public, you want to restrict the public from certain things, like courses, because it most likely going to be a paid course, and even if they weren't, you would only want authenticated users to access the courses, other things we would want to restrict the public from would be everything except the enrollment, we could allow access to contact and chat interfaces if there was. 
```

Comment:
```suggestion
You want to be careful with those you give administrator roles to, as it has access to administer all data, and restrict the public role from certain things, like courses enrollment, because you may only want authenticated users to access them.
```